freqtrade/freqtradebot.py

```diff
@@ -443,6 +448,49 @@ def create_trade(self, pair: str) -> bool:
         else:
             return False
 
+#
+# BUY / increase positions / DCA logic and methods
+#
+    def process_open_trade_positions(self):
+        """
+        Tries to execute additional buy or sell orders for open trades (positions)
+        """
+        # Walk through each pair and check if it needs changes
+        for trade in Trade.get_open_trades():
+            # If there is any open orders, wait for them to finish.
+            for order in trade.orders:
+                if order.ft_is_open:
+                    break
+            try:
+                self.check_and_call_adjust_trade_position(trade)
+            except DependencyException as exception:
+                logger.warning('Unable to adjust position of trade for %s: %s',
+                               trade.pair, exception)
+
+    def check_and_call_adjust_trade_position(self, trade: Trade):
+        """
+        Check the implemented trading strategy for adjustment command.
+        If the strategy triggers the adjustment, a new order gets issued.
+        Once that completes, the existing trade is modified to match new data.
+        """
+        current_rate = self.exchange.get_rate(trade.pair, refresh=True, side="buy")
```

Comment:
This still leaves me somewhat doubful.

Assuming a lowish value pair (-> a high spread) - we could see the following (this assumes reversed price sides):
Telegram/api reports a profit of -5% (assuming sell rate)
`adjust_trade_position()` will be called with a profit of -4.5% (or even 4%)

Now obviously, for buying again, we'd want to use the "buy" side price
However for determining the current profit to decide on DCA, i'm not entirely sure which price you'd want to look at - and it'll strongly depend on the pricing configuration which one will actually make sense.

if we don't wanna have this "fixed" to either buy or sell - maybe we can leave the user in the driver seat and have a "pricing_side_DCA" (naming might be improved) - which can default to buy (or sell) - but gives the user/strategy the last word in it?

Reply:
Would it suffice if we change the method signature documentation to be?
:param current_rate: Current buy rate. Use `exchange.get_rate` if you need sell rate.

In this way, we leave a guide for those who need it but get to keep the signature the same as for all other callbacks.

Reply:
the strategy does not have direct access to `exchange` - nor should it be directly exposed (you can lock your bot up (or crash it) with some ill-placed calls to the exchange class).

If we need this, it should be through an interface in `dataprovider` (so it'd be `dp.get_rate`).